c-dependencies/js-compute-runtime/js-compute-builtins.h

```diff
@@ -85,6 +85,107 @@ bool define_fastly_sys(JSContext *cx, JS::HandleObject global);
 
 bool RejectPromiseWithPendingError(JSContext *cx, JS::HandleObject promise);
 
+namespace URL {
+bool is_instance(JS::Value val);
+}
+
+namespace FetchEvent {
+enum class State {
+  unhandled,
+  waitToRespond,
+  responseStreaming,
+  responseDone,
+  responsedWithError,
+};
+
+JSObject *create(JSContext *cx);
+bool init_request(JSContext *cx, JS::HandleObject self);
+
+// There can only ever be a single FetchEvent instance in a service, so we can
+// treat it as a singleton for easy access. Returns a nullptr if the FetchEvent
+// hasn't been created yet.
+JS::HandleObject instance();
+
+State state(JSObject *self);
+void set_state(JSObject *self, State state);
+
+// https://w3c.github.io/ServiceWorker/#extendableevent-active
+bool is_active(JSObject *self);
+
+bool is_dispatching(JSObject *self);
+void start_dispatching(JSObject *self);
+void stop_dispatching(JSObject *self);
+
+bool response_started(JSObject *self);
+bool respondWithError(JSContext *cx, JS::HandleObject self);
+} // namespace FetchEvent
+
+namespace GlobalProperties {
+// Maps an encoded character to a value in the Base64 alphabet, per
+// RFC 4648, Table 1. Invalid input characters map to UINT8_MAX.
+// https://datatracker.ietf.org/doc/html/rfc4648#section-4
+
+constexpr uint8_t nonAlphabet = 255;
+
+static const uint8_t base64DecodeTable[] = {
```

Comment:
Could you move this back to `js-compute-builtins.cpp` and make this declaration `extern const uint8_t base64DecodeTable[];` instead?